### PR TITLE
Fix TaskAttempt Context is null

### PR DESCRIPTION
**Pull request by Vipinofficial11**
Changes include:

1. Fixes in the ```SalesforceWideRecordReader``` partitioning logic to address API retrieve limit exceeded errors.
2. Usage of default configurations in case of Non-CDAP calls

### CODE REVIEW

**Comment by MrRahulSharma** [comment]
Can we validate the WideRecordReader changes in CDAP/CDF today ?

File: src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceWideRecordReader.java [hunk 1]
```diff
@@ -70,7 +70,7 @@ public void initialize(InputSplit inputSplit, TaskAttemptContext taskAttemptCont
   public SalesforceWideRecordReader initialize(
       InputSplit inputSplit, AuthenticatorCredentials credentials)
       throws IOException, InterruptedException {
-    List<Map<String, ?>> fetchedIdList = fetchBulkQueryIds(inputSplit, null);
+    List<Map<String, ?>> fetchedIdList = fetchBulkQueryIds(inputSplit, credentials);
```
Comment (MrRahulSharma) [inline src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceWideRecordReader.java]: Move the super.initialize call outside fetchBulkQueryIds and call it before the fetchBulkQueryIds call here.

Comment (Vipinofficial11) [reply src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceWideRecordReader.java]: Moved ```super.initialize```  before ```fetchBulkQueryIds``` and removed paramters from ```fetchBulkQueryIds``` as they are not required anymore in the method. [b5c9fe5d483b43f874cdf4ade9370479ca92b1c0]

**Review by MrRahulSharma — APPROVED** [review]
LGTM

**Pull request merged**